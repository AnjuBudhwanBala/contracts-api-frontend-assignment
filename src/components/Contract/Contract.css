.ContractItem {
  display: inline-block;
  width: 20%;
  vertical-align: top;
}
@media screen and (max-width: 1024px) {
  .ContractItem {
    width: 25%;
  }
}
@media screen and (max-width: 768px) {
  .ContractItem {
    width: 50%;
  }
}
@media screen and (max-width: 600px) {
  .ContractItem {
    width: 100%;
  }
}
.CompanyName {
  width: 100%;
  text-align: center;
}
.CompanyIcon {
  background: rgb(0, 132, 132);
  height: 50px;
  width: 50px;
  border-radius: 50%;
  color: #fff;
  position: relative;
  margin: 10px auto;
}
.CompanyIcon i {
  text-align: center;
  position: absolute;
  left: 50%;
  right: 0px;
  top: 50%;
  transform: translate(-50%, -50%);
}
.ContractDetails {
  padding: 30px;
  margin: 10px;
  box-shadow: 2px 2px 10px 1px #ccc;
  text-align: left;
  background: #fff;
  min-height: 360px;
  max-height: 400px;
  position: relative;
}

.ContractDetails ul {
  list-style: none;
  padding: 0;
  margin: 0;
}

.ContractDetails li {
  list-style: none;
  padding: 10px 0;
  border-bottom: 1px solid #ccc;
}
.ContractDetails li:first-child {
  color: rgb(0, 132, 132);
  text-align: center;
  font-weight: bold;
  margin-bottom: 20px;
  border: none;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.disabledText {
  color: #888;
  font-size: 12px;
  margin-top: 20px;
}

.ContractDetails .CustomButton {
  position: absolute;
  bottom: 20px;
  left: 50%;
  transform: translate(-50%);
}
