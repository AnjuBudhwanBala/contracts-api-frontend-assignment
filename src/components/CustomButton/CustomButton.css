.CustomButton {
  text-align: center;
}

.CustomButton button {
  color: #fff;
  background: rgb(0, 132, 132);
  border: none;
  padding: 8px 30px;
  margin: 20px auto;
  opacity: 0.6;
  transition: 0.3s;
}
.CustomButton button:hover {
  box-shadow: 4px 4px 10px #ccc;
  border-radius: 2px;
  opacity: 1;
}
