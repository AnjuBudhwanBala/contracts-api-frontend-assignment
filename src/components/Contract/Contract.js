import React, { useState } from 'react';
import 'font-awesome/css/font-awesome.min.css';
import './Contract.css';
import CustomButton from '../CustomButton/CustomButton';
import Modal from '../../components/Modal/Modal';
import CustomDateConverter from '../CustomDateConverter/CustomDateConverter';

const Contract = ({ contractInfo }) => {
  const [modalState, setModalState] = useState(false);
  const {
    company,
    negotiationRenewalDate,
    periodEnd,
    periodStart,
    scheduleForRenewal
  } = contractInfo;

  const openModalHandler = () => {
    setModalState(true);
  };
  const closeModalHandler = () => {
    setModalState(false);
  };

  return (
    <div className="ContractItem">
      <div className="ContractDetails">
        <div className="CompanyName">
          <div className="CompanyIcon">
            <i className="fa fa-industry" aria-hidden="true"></i>
          </div>
        </div>
        <ul>
          <li>
            <span>{company} </span>
          </li>
          <li>
            <strong>Start : </strong>
            <span>
              <CustomDateConverter date={periodStart} />
            </span>
          </li>
          <li>
            <strong>End : </strong>
            <span>
              <CustomDateConverter date={periodEnd} />
            </span>
          </li>
          {scheduleForRenewal ? (
            <li>
              <strong> Renewal Date : </strong>
              <span>
                <CustomDateConverter date={negotiationRenewalDate} />
              </span>
            </li>
          ) : (
            <span className="disabledText">Not schedule for renewal</span>
          )}
        </ul>
        <CustomButton
          click={openModalHandler}
          disabled={modalState}
          btnType="button"
        >
          Edit
        </CustomButton>
      </div>
      {modalState ? (
        <Modal
          contractInfo={contractInfo}
          closeModal={closeModalHandler}
          modalState={modalState}
        ></Modal>
      ) : null}
    </div>
  );
};

export default Contract;
