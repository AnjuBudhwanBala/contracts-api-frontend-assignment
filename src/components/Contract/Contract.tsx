import React, { useState } from 'react';
import 'font-awesome/css/font-awesome.min.css';
import './Contract.css';
import CustomButton from '../CustomButton/CustomButton';
import Modal from '../Modal/Modal';
import CustomDateConverter from '../CustomDateConverter/CustomDateConverter';
import Form from '../Form/Form';

export interface ContractObj {
  company: string;
  negotiationRenewalDate: string;
  periodEnd: string;
  periodStart: string;
  scheduleForRenewal: boolean;
  contractId?: string;
}

interface Props {
  contractInfo: ContractObj;
  refreshContracts: () => void;
}

export const Contract: React.FC<Props> = props => {
  const { contractInfo, refreshContracts } = props;

  const [isModalOpen, setIsModalOpen] = useState(false);
  const {
    company,
    negotiationRenewalDate,
    periodEnd,
    periodStart,
    scheduleForRenewal
  } = contractInfo;

  // to open Modal
  const openModalHandler = () => {
    setIsModalOpen(true);
  };
  //to close Modal
  const closeModalHandler = () => {
    setIsModalOpen(false);
  };
  //update contract after refresh
  const updateContract = () => {
    refreshContracts();
  };

  return (
    <div className="ContractItem">
      <div className="ContractDetails">
        <div className="CompanyName">
          <div className="CompanyIcon">
            <i className="fa fa-industry" aria-hidden="true"></i>
          </div>
        </div>
        <ul>
          <li>
            <span>{company}</span>
          </li>
          <li>
            <strong>Start : </strong>
            <span>
              <CustomDateConverter date={periodStart} />
            </span>
          </li>
          <li>
            <strong>End : </strong>
            <span>
              <CustomDateConverter date={periodEnd} />
            </span>
          </li>
          {scheduleForRenewal ? (
            <li>
              <strong> Renewal Date : </strong>
              <span>
                <CustomDateConverter date={negotiationRenewalDate} />
              </span>
            </li>
          ) : (
            <span
              style={{ color: '#888', fontSize: '12px', marginTop: '20px' }}
            >
              Not schedule for renewal
            </span>
          )}
        </ul>
        <CustomButton click={openModalHandler} btnType="button">
          Edit
        </CustomButton>
      </div>
      {isModalOpen ? (
        <Modal closeModal={closeModalHandler} modalState={isModalOpen}>
          <Form
            contractInfo={contractInfo}
            updateContract={updateContract}
            closeModal={closeModalHandler}
          ></Form>
        </Modal>
      ) : null}
    </div>
  );
};

export default Contract;
