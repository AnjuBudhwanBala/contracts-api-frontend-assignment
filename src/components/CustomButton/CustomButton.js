import React from 'react';
import './CustomButton.css';

const CustomButton = ({ click, children, btnType, isDisabled }) => {
  return (
    <div className="CustomButton">
      <button type={btnType} onClick={click} disabled={isDisabled}>
        {children}
      </button>
    </div>
  );
};

export default CustomButton;
