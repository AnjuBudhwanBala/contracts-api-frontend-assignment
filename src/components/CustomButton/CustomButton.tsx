import React from 'react';
import './CustomButton.css';

interface Props {
  click?: () => void;
  children: React.ReactNode;
  btnType: 'button' | 'submit' | 'reset' | undefined;
  isDisabled?: boolean;
}

const CustomButton: React.FC<Props> = props => {
  const { click, children, btnType, isDisabled } = props;
  return (
    <div className="CustomButton">
      <button type={btnType} onClick={click} disabled={isDisabled}>
        {children}
      </button>
    </div>
  );
};

export default CustomButton;
