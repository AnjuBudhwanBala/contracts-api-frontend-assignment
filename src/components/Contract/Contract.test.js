import React from 'react';
import { shallow } from 'enzyme';
import Contract from './Contract';
import { renderHook, act } from '@testing-library/react-hooks';
import CustomButton from '../CustomButton/CustomButton';

describe('Contract Component', () => {
  let wrapper;
  let mockRefreshContracts;

  beforeEach(() => {
    mockRefreshContracts = jest.fn();

    const mockProps = {
      contractInfo: {
        company: 'abc'
      },
      refreshContracts: mockRefreshContracts
    };

    wrapper = shallow(<Contract {...mockProps} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render Contract component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
