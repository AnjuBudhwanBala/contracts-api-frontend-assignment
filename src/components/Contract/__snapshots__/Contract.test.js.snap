// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contract Component should render Contract component 1`] = `ShallowWrapper {}`;
